interfaces/builtin/opengl.go

```diff
@@ -110,6 +110,10 @@ unix (bind,listen) type=seqpacket addr="@cuda-uvmfd-[0-9a-f]*",
 /dev/tegra_dc_ctrl rw,
 /dev/tegra_dc_[0-9]* rw,
 
+# Xilinx zocl DRM driver
+# https://github.com/Xilinx/XRT/tree/master/src/runtime_src/core/edge/drm
+/sys/devices/platform/amba_pl@[0-9]*/amba_pl@[0-9]*:zyxclmm_drm/* r,
```

Comment:
from the linked repo, they mention this path:

should this be 

```suggestion
/sys/bus/platform/devices/amba_pl@[0-9]*/amba_pl@[0-9]*:zyxclmm_drm/* r,
```

?

see https://github.com/Xilinx/XRT/blob/e3e0992d275cc2109b3f533ee36ea71b24ffbbb0/src/runtime_src/doc/toc/sysfs.rst

if you have tested with this specific path and found that this specific rule is necessary, can you provide an example denial?

Reply:
It needs to be `/sys/devices` and not `/sys/bus/platform/devices`. The reason is that apparmor looks at the real path, and the latter is actually a symlink to the former:
```
ubuntu@zynqmp:/sys$ ls -ld devices/platform/amba_pl@0/
drwxr-xr-x 17 root root 0 Feb 25 07:36 devices/platform/amba_pl@0/
ubuntu@zynqmp:/sys$ ls -ld bus/platform/devices/amba_pl@0
lrwxrwxrwx 1 root root 0 Feb 25 07:36 bus/platform/devices/amba_pl@0 -> ../../../devices/platform/amba_pl@0
```
This is the denial without the rule (or, with the `/sys/bus/platform/devices` one):
```
[  372.181087] audit: type=1400 audit(1614238934.996:77): apparmor="DENIED" operation="open" profile="snap.abeato-dpu-facedetect-demo.facedetect" name="/sys/devices/platform/amba_pl@0/amba_pl@0:zyxclmm_drm/ip_layout" pid=2564 comm="test_jpeg_faced" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
```

Reply:
ah right thanks for the clarification, I think we still need a slight tweak to your rule:

```suggestion
/sys/devices/platform/amba_pl@[0-9]+/amba_pl@[0-9]+:zyxclmm_drm/* r,
```

this is to prevent access to something like 
```
/sys/devices/platform/amba_pl@/amba_pl@:zyxclmm_drm/*
```

Reply:
@anonymouse64 your suggestion actually does not avoid the denial, because iirc apparmor uses globbing, not full regular expressions, so that `+` is a literal in this case.

Reply:
sorry I always forget that apparmor RE is not actual RE, this is fine then